apps/system/lockscreen/js/lockscreen_state_manager.js

```diff
@@ -411,14 +413,46 @@
       case 'lockscreen-notify-homepressed':
         this.onHomePressed();
         break;
-      case 'lockscreenslide-activate-left':
-        this.onUnlockingApp();
-        break;
-      case 'lockscreenslide-activate-right':
       case 'lockscreen-notification-request-activate-unlock':
         this.onActivateUnlock();
+        isFromNotification = true;
+        appID = detail.id;
         break;
       case 'lockscreen-request-unlock':
+      // if app is launched from the lockscreen notification screen then it will
```

Comment:
I agree with David at this point - this code shouldn't even be here.
launch of an application currently performed in 'visibility_manager.js' as a response to data provided with ``lockscreen-request-unlock`` event.